caddyhttp/httpserver/replacer.go

```diff
@@ -95,6 +104,15 @@ func NewReplacer(r *http.Request, rr *ResponseRecorder, emptyValue string) Repla
 				dir, _ := path.Split(r.URL.Path)
 				return dir
 			}(),
+			"{request}": func() string {
+				dump, err := httputil.DumpRequest(r, false)
+				if err != nil {
+					// should we return err.Error()?
+					return ""
```

Comment:
> // should we return err.Error()?

Only if you want the error showing up where you don't expect it. :wink: A quick glance inside the DumpRequest() function with false as the second argument reveals that an error is very unlikely... the system would have to be completely out of memory I think. So I think returning empty string is fine.